scripts/detect_text.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (C) 2021-2022, Mindee.
+
+# This program is licensed under the Apache License version 2.
+# See LICENSE or go to <https://www.apache.org/licenses/LICENSE-2.0.txt> for full license details.
+
+import argparse
+import json
+import os
+from pathlib import Path
+
+from tqdm import tqdm
+
+os.environ["TF_CPP_MIN_LOG_LEVEL"] = "2"
+
+from doctr.file_utils import is_tf_available
+from doctr.io import DocumentFile
+from doctr.models import ocr_predictor
+
+# Enable GPU growth if using TF
+if is_tf_available():
+    import tensorflow as tf
+
+    gpu_devices = tf.config.experimental.list_physical_devices("GPU")
+    if any(gpu_devices):
+        tf.config.experimental.set_memory_growth(gpu_devices[0], True)
+
+
+_OUTPUT_CHOICE_JSON = "json"
+_OUTPUT_CHOICE_TEXT = "txt"
+
+
+def _process_file(model, file_path: Path, out_format: str) -> str:
+    if str(file_path).lower().endswith(".pdf"):
+        doc = DocumentFile.from_pdf(file_path)
+    else:
+        doc = DocumentFile.from_images(file_path)
+
+    out = model(doc)
+    export = out.export()
+
+    if out_format == _OUTPUT_CHOICE_JSON:
+        out_txt = json.dumps(export, indent=2)
+    elif out_format == _OUTPUT_CHOICE_TEXT:
+        out_txt = ""
+        for page in export["pages"]:
+            for block in page["blocks"]:
+                for line in block["lines"]:
+                    for word in line["words"]:
+                        out_txt += word["value"] + " "
```

Comment:
Perhaps
```python
out_txt += " ".join(word["value"] for word in line["words"])
```

or wrapping more nested loops inside a list comprehensions :man_shrugging:

Reply:
out.render() ?

Reply:
Oh yeah actually, I had forgotten about this haha